seek_to.c

```diff
@@ -8,7 +8,7 @@
 SeekResult leveldb_iter_seek_to(leveldb_iterator_t* iter, const char* k, size_t klen) {
     leveldb_iter_seek(iter, k, klen);
 
-    SeekResult sr;
+    SeekResult sr = {};
     sr.valid = leveldb_iter_valid(iter);
```

Comment:
hum, I always used `memset(&obj, 0, sizeof(obj))`. Didn't know `{}` actually sets all the field bits to zero...

Reply:
Thanks for the comment. Upon further review, I realized I was using syntax that was valid for C++ but not for C (for C++ this would be aggregate initialization with an empty init list: https://en.cppreference.com/w/cpp/language/initialization).

Using `= {0}` is valid C syntax, however, and easier to read than `memset`. For posterity, here's why that works:

The syntax we are invoking is struct initialization:
https://en.cppreference.com/w/c/language/struct_initialization

Relevant quotes:
"When initializing a struct, the first initializer in the list initializes the first declared member"
"All members that are not initialized explicitly are initialized implicitly the same way as objects that have static storage duration."

Static initialization initializes integers to zero:
https://en.cppreference.com/w/c/language/initialization

"objects with static and thread-local storage duration are initialized as follow ... objects of integral types are initialized to unsigned zero"

When I used GodBolt, the empty braces did the correct thing, but `-pedantic` gave me a warning: https://godbolt.org/z/7aieQu

Reply:
nice, TIL `{}` gives a warning in C.